#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_livelock.dpatch by Timothy Baldwin <T.E.Baldwin99@members.leeds.ac.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix a livelock problem when jumping between editing marks.

@DPATCH@
diff -urNad vdr-1.3.23/dvbplayer.c /tmp/dpep.KvgOVM/vdr-1.3.23/dvbplayer.c
--- vdr-1.3.23/dvbplayer.c	2005-04-12 01:28:23.392793314 +0100
+++ /tmp/dpep.KvgOVM/vdr-1.3.23/dvbplayer.c	2005-04-12 01:28:23.978745831 +0100
@@ -464,6 +464,7 @@
         cPoller Poller;
         if (DevicePoll(Poller, 100)) {
 
+           sched_yield();
            LOCK_THREAD;
 
            // Read the next frame from the file:
